### PR TITLE
UserProfile resource

### PR DESCRIPTION
**Pull request by akartsky**
UserProfile resource

Take a look at following files :
  - generator.yaml
  - hook.go
  - template/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: helm/templates/deployment.yaml [hunk 1]
```diff
@@ -61,6 +61,8 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
+        - name: AWS_ACCOUNT_ID
```
Comment (ryansteakley) [inline helm/templates/deployment.yaml:64]: same as others

File: helm/values.yaml [hunk 2]
```diff
@@ -38,6 +38,7 @@ resources:
 aws:
   # If specified, use the AWS region for AWS API calls
   region: ""
+  account_id: ""
```
Comment (ryansteakley) [inline helm/values.yaml:41]: same as others

File: pkg/resource/user_profile/hook.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+// requeueUntilCanModify creates and returns an
+// ackrequeue error if a resource's latest status matches
+// any of the defined modifying statuses below.
+func (rm *resourceManager) requeueUntilCanModify(ctx context.Context, r *resource) error {
+	latestStatus := r.ko.Status.Status
+	return svccommon.RequeueIfModifying(latestStatus, &resourceName, &modifyingStatuses)
+}
+
+// customDescribeUserProfileSetOutput sets the resource ResourceSynced condition to False if endpoint is
```
Comment (ryansteakley) [inline pkg/resource/user_profile/hook.go:49]: should this be userprofile and not endpoint

**Review by ryansteakley — APPROVED** [review]

File: pkg/resource/user_profile/resource.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+// SetStatus will set the Status field for the resource
+func (r *resource) SetStatus(desired acktypes.AWSResource) {
+	r.ko.Status = desired.(*resource).ko.Status
+}
+
+// SetIdentifiers sets the Spec or Status field that is referenced as the unique
+// resource identifier
+func (r *resource) SetIdentifiers(identifier *ackv1alpha1.AWSIdentifiers) error {
```
Comment (mbaijal) [inline pkg/resource/user_profile/resource.go:94]: Verified that it looks correct.

**Review by mbaijal — APPROVED** [review]

File: pkg/resource/user_profile/resource.go (hunk 4 above)
Comment (akartsky) [reply pkg/resource/user_profile/resource.go:94]: thanks !

**Pull request merged**